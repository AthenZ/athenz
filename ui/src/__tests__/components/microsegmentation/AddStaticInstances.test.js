/*
 * Copyright The Athenz Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'react';
import {fireEvent, render} from '@testing-library/react';
import AddStaticinstances from "../../../components/microsegmentation/AddStaticInstances";
import API from "../../../api";

describe('AddStaticInstances', () => {
    it('should render', () => {
        let domain = 'domain';
        const showAddStaticinstances = true;
        const cancel = function() {};
        const submit = function() {};
        let _csrf = 'csrf';

        const { getByTestId } = render(
            <AddStaticinstances
                api={API()}
                domain={domain}
                onSubmit={submit}
                onCancel={cancel}
                _csrf={_csrf}
                showAddInstance={showAddStaticinstances}
                justificationRequired={false}
            />
        );
        const addinstances = getByTestId('add-modal-message');
        expect(addinstances).toMatchSnapshot();
    });

    it('should render fail to submit add-segmentation: ResourceType is required', () => {
        const showAddStaticinstances = true;
        const cancel = function() {};
        const domain = 'domain';
        let role ='roleName';
        const submit = function() {};
        let _csrf = 'csrf';;
        const { getByTestId, getByText } =  render(
            <AddStaticinstances
                api={API()}
                domain={domain}
                onSubmit={submit}
                onCancel={cancel}
                _csrf={_csrf}
                showAddInstance={showAddStaticinstances}
            />
        );
        const addStaticinstances = getByTestId('add-modal-message');
        expect(addStaticinstances).toMatchSnapshot();

        fireEvent.click(getByText('Submit'));

        expect(getByText('Resource Type is required.')).not.toBeNull();
    });

});
