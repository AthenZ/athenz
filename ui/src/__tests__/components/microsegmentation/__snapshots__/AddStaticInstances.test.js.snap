// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddStaticInstances should render 1`] = `
.emotion-0 {
  text-align: left;
  font: 300 14px HelveticaNeue-Reg,Helvetica,Arial,sans-serif;
  padding-bottom: 15px;
  max-height: auto;
  overflow-y: scroll;
}

.emotion-2 {
  width: 780px;
  text-align: left;
  background-color: #fff;
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  padding: 10px 10px;
}

.emotion-6 {
  font-size: 14px;
  font-weight: 600;
  color: #303030;
  line-height: 36px;
  white-space: nowrap;
}

.emotion-7 {
  float: left;
  font-size: 14px;
  font-weight: 700;
  padding-top: 12px;
  width: 20%;
}

.emotion-9 {
  position: relative;
}

.emotion-9 input {
  cursor: inherit;
}

.emotion-10 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 14px;
  font-weight: 300;
  position: relative;
  width: 250px;
}

.emotion-10 input {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: rgba(53,112,244,0.05);
  border-top: 2px solid transparent;
  border-bottom: 2px solid transparent;
  border-right: transparent;
  border-left: transparent;
  border-radius: 2px;
  box-shadow: none;
  box-sizing: border-box;
  color: #303030;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  font: inherit;
  height: 36px;
  margin: 0;
  outline: none;
  padding: 0 1rem;
  text-align: left;
  width: 100%;
}

.emotion-10.animated input {
  -webkit-transition: background-color 0.2s ease-in-out,color 0.2s ease-in-out,border 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out,color 0.2s ease-in-out,border 0.2s ease-in-out;
}

.emotion-10 input::-webkit-input-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-10 input::-moz-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-10 input:-ms-input-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-10 input::placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-10 input.focused,
.emotion-10 input:active,
.emotion-10 input:focus {
  background: rgba(53,112,244,0.05);
  border-bottom: 2px solid #3570f4;
  color: #303030;
}

.emotion-10 input:invalid {
  background: rgba(53,112,244,0.05);
  border-bottom: 2px solid #d01111;
  color: #303030;
}

.emotion-10 .message {
  font-size: 12px;
  left: 0;
  line-height: 1.8;
  position: absolute;
  top: 2.571rem;
}

.emotion-10.error input,
.emotion-10.error input.focused,
.emotion-10.error input:active,
.emotion-10.error input:focus {
  border-bottom: 2px solid #d01111;
}

.emotion-10.error .message {
  color: #d01111;
}

.emotion-10 input {
  padding-right: 36px;
}

.emotion-10 .input-icon {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 36px;
  position: absolute;
  right: 6px;
}

.emotion-11 {
  fill: #d5d5d5;
  cursor: pointer;
  vertical-align: text-bottom;
}

.emotion-12 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 14px;
  font-weight: 300;
  position: relative;
  width: 100%;
}

.emotion-12 input {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: rgba(53,112,244,0.05);
  border-top: 2px solid transparent;
  border-bottom: 2px solid transparent;
  border-right: transparent;
  border-left: transparent;
  border-radius: 2px;
  box-shadow: none;
  box-sizing: border-box;
  color: #303030;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  font: inherit;
  height: 36px;
  margin: 0;
  outline: none;
  padding: 0 1rem;
  text-align: left;
  width: 100%;
}

.emotion-12.animated input {
  -webkit-transition: background-color 0.2s ease-in-out,color 0.2s ease-in-out,border 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out,color 0.2s ease-in-out,border 0.2s ease-in-out;
}

.emotion-12 input::-webkit-input-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-12 input::-moz-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-12 input:-ms-input-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-12 input::placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-12 input.focused,
.emotion-12 input:active,
.emotion-12 input:focus {
  background: rgba(53,112,244,0.05);
  border-bottom: 2px solid #3570f4;
  color: #303030;
}

.emotion-12 input:invalid {
  background: rgba(53,112,244,0.05);
  border-bottom: 2px solid #d01111;
  color: #303030;
}

.emotion-12 .message {
  font-size: 12px;
  left: 0;
  line-height: 1.8;
  position: absolute;
  top: 2.571rem;
}

.emotion-12.error input,
.emotion-12.error input.focused,
.emotion-12.error input:active,
.emotion-12.error input:focus {
  border-bottom: 2px solid #d01111;
}

.emotion-12.error .message {
  color: #d01111;
}

.emotion-13 {
  width: 350px;
  padding: 0 12px;
}

.emotion-15 {
  margin-top: 5px;
  margin-left: 5px;
}

.emotion-17 {
  fill: #188fff;
  cursor: pointer;
  vertical-align: text-bottom;
}

.emotion-23 {
  width: 65%;
}

<div
  class="emotion-0 emotion-1"
  data-testid="add-modal-message"
  height="auto"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <label
        class="emotion-6 denali-input-label emotion-7 emotion-8"
      >
        Type and value:
      </label>
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="downshift-0-label"
        class="emotion-9 denali-input-dropdown"
        data-testid="input-dropdown"
        role="combobox"
      >
        <div
          class="emotion-10 denali-input"
          data-testid="input-wrapper"
        >
          <input
            aria-autocomplete="list"
            aria-labelledby="downshift-0-label"
            autocomplete="off"
            autocorrect="off"
            data-testid="input-node"
            id="downshift-0-input"
            name="resourceType"
            placeholder="Select Resourcetype"
            spellcheck="false"
            value=""
          />
          <div
            class="input-icon"
          >
            
            <svg
              class="emotion-11"
              data-testid="icon"
              height="24px"
              viewBox="0 0 1024 1024"
              width="24px"
            >
              <title>
                arrowhead-down
              </title>
              <path
                d="M512 580.267l-226.133-226.133c-7.461-8.401-18.291-13.668-30.351-13.668-13.957 0-26.267 7.055-33.558 17.793l-0.091 0.142c-5.319 7.276-8.51 16.4-8.51 26.27 0 12.199 4.876 23.259 12.785 31.338l-0.008-0.008 256 253.867c7.434 8.083 18.061 13.13 29.867 13.13s22.432-5.047 29.84-13.101l0.026-0.029 256-253.867c7.901-8.071 12.777-19.131 12.777-31.33 0-9.87-3.192-18.994-8.599-26.397l0.088 0.127c-7.382-10.88-19.691-17.934-33.649-17.934-12.060 0-22.89 5.267-30.315 13.626l-0.036 0.042z"
              />
            </svg>
          </div>
          <div
            class="message"
            data-testid="message"
          />
        </div>
      </div>
      <div
        class="emotion-12 denali-input emotion-13 emotion-14"
        data-testid="input-wrapper"
      >
        <input
          autocomplete="off"
          data-testid="input-node"
          placeholder="Enter value"
          value=""
        />
        <div
          class="message"
          data-testid="message"
        />
      </div>
      <div
        class="emotion-15 emotion-16"
      >
        <svg
          class="emotion-17"
          data-testid="icon"
          height="1.75em"
          viewBox="0 0 1024 1024"
          width="1.75em"
        >
          <title>
            add-circle
          </title>
          <path
            d="M512 42.667c-259.206 0-469.333 210.128-469.333 469.333s210.128 469.333 469.333 469.333c259.206 0 469.333-210.128 469.333-469.333v0c0.003-0.635 0.005-1.387 0.005-2.138 0-258.027-209.173-467.2-467.2-467.2-0.752 0-1.504 0.002-2.255 0.005l0.117-0zM512 896c-212.077 0-384-171.923-384-384s171.923-384 384-384c212.077 0 384 171.923 384 384v0c-1.203 211.592-172.408 382.797-383.885 383.999l-0.115 0.001z"
          />
          <path
            d="M725.333 469.333h-170.667v-170.667c0-23.564-19.103-42.667-42.667-42.667s-42.667 19.103-42.667 42.667v0 170.667h-170.667c-23.564 0-42.667 19.103-42.667 42.667s19.103 42.667 42.667 42.667v0h170.667v170.667c0 23.564 19.103 42.667 42.667 42.667s42.667-19.103 42.667-42.667v0-170.667h170.667c23.564 0 42.667-19.103 42.667-42.667s-19.103-42.667-42.667-42.667v0z"
          />
        </svg>
      </div>
    </div>
    <div
      class="emotion-4 emotion-5"
    >
      <label
        class="emotion-6 denali-input-label emotion-7 emotion-8"
      />
      <div
        class="emotion-23 emotion-24"
      />
    </div>
  </div>
</div>
`;

exports[`AddStaticInstances should render fail to submit add-segmentation: ResourceType is required 1`] = `
.emotion-0 {
  text-align: left;
  font: 300 14px HelveticaNeue-Reg,Helvetica,Arial,sans-serif;
  padding-bottom: 15px;
  max-height: auto;
  overflow-y: scroll;
}

.emotion-2 {
  width: 780px;
  text-align: left;
  background-color: #fff;
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  padding: 10px 10px;
}

.emotion-6 {
  font-size: 14px;
  font-weight: 600;
  color: #303030;
  line-height: 36px;
  white-space: nowrap;
}

.emotion-7 {
  float: left;
  font-size: 14px;
  font-weight: 700;
  padding-top: 12px;
  width: 20%;
}

.emotion-9 {
  position: relative;
}

.emotion-9 input {
  cursor: inherit;
}

.emotion-10 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 14px;
  font-weight: 300;
  position: relative;
  width: 250px;
}

.emotion-10 input {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: rgba(53,112,244,0.05);
  border-top: 2px solid transparent;
  border-bottom: 2px solid transparent;
  border-right: transparent;
  border-left: transparent;
  border-radius: 2px;
  box-shadow: none;
  box-sizing: border-box;
  color: #303030;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  font: inherit;
  height: 36px;
  margin: 0;
  outline: none;
  padding: 0 1rem;
  text-align: left;
  width: 100%;
}

.emotion-10.animated input {
  -webkit-transition: background-color 0.2s ease-in-out,color 0.2s ease-in-out,border 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out,color 0.2s ease-in-out,border 0.2s ease-in-out;
}

.emotion-10 input::-webkit-input-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-10 input::-moz-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-10 input:-ms-input-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-10 input::placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-10 input.focused,
.emotion-10 input:active,
.emotion-10 input:focus {
  background: rgba(53,112,244,0.05);
  border-bottom: 2px solid #3570f4;
  color: #303030;
}

.emotion-10 input:invalid {
  background: rgba(53,112,244,0.05);
  border-bottom: 2px solid #d01111;
  color: #303030;
}

.emotion-10 .message {
  font-size: 12px;
  left: 0;
  line-height: 1.8;
  position: absolute;
  top: 2.571rem;
}

.emotion-10.error input,
.emotion-10.error input.focused,
.emotion-10.error input:active,
.emotion-10.error input:focus {
  border-bottom: 2px solid #d01111;
}

.emotion-10.error .message {
  color: #d01111;
}

.emotion-10 input {
  padding-right: 36px;
}

.emotion-10 .input-icon {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 36px;
  position: absolute;
  right: 6px;
}

.emotion-11 {
  fill: #d5d5d5;
  cursor: pointer;
  vertical-align: text-bottom;
}

.emotion-12 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 14px;
  font-weight: 300;
  position: relative;
  width: 100%;
}

.emotion-12 input {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: rgba(53,112,244,0.05);
  border-top: 2px solid transparent;
  border-bottom: 2px solid transparent;
  border-right: transparent;
  border-left: transparent;
  border-radius: 2px;
  box-shadow: none;
  box-sizing: border-box;
  color: #303030;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  font: inherit;
  height: 36px;
  margin: 0;
  outline: none;
  padding: 0 1rem;
  text-align: left;
  width: 100%;
}

.emotion-12.animated input {
  -webkit-transition: background-color 0.2s ease-in-out,color 0.2s ease-in-out,border 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out,color 0.2s ease-in-out,border 0.2s ease-in-out;
}

.emotion-12 input::-webkit-input-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-12 input::-moz-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-12 input:-ms-input-placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-12 input::placeholder {
  color: rgba(48,48,48,0.6);
}

.emotion-12 input.focused,
.emotion-12 input:active,
.emotion-12 input:focus {
  background: rgba(53,112,244,0.05);
  border-bottom: 2px solid #3570f4;
  color: #303030;
}

.emotion-12 input:invalid {
  background: rgba(53,112,244,0.05);
  border-bottom: 2px solid #d01111;
  color: #303030;
}

.emotion-12 .message {
  font-size: 12px;
  left: 0;
  line-height: 1.8;
  position: absolute;
  top: 2.571rem;
}

.emotion-12.error input,
.emotion-12.error input.focused,
.emotion-12.error input:active,
.emotion-12.error input:focus {
  border-bottom: 2px solid #d01111;
}

.emotion-12.error .message {
  color: #d01111;
}

.emotion-13 {
  width: 350px;
  padding: 0 12px;
}

.emotion-15 {
  margin-top: 5px;
  margin-left: 5px;
}

.emotion-17 {
  fill: #188fff;
  cursor: pointer;
  vertical-align: text-bottom;
}

.emotion-23 {
  width: 65%;
}

<div
  class="emotion-0 emotion-1"
  data-testid="add-modal-message"
  height="auto"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <label
        class="emotion-6 denali-input-label emotion-7 emotion-8"
      >
        Type and value:
      </label>
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="downshift-1-label"
        class="emotion-9 denali-input-dropdown"
        data-testid="input-dropdown"
        role="combobox"
      >
        <div
          class="emotion-10 denali-input"
          data-testid="input-wrapper"
        >
          <input
            aria-autocomplete="list"
            aria-labelledby="downshift-1-label"
            autocomplete="off"
            autocorrect="off"
            data-testid="input-node"
            id="downshift-1-input"
            name="resourceType"
            placeholder="Select Resourcetype"
            spellcheck="false"
            value=""
          />
          <div
            class="input-icon"
          >
            
            <svg
              class="emotion-11"
              data-testid="icon"
              height="24px"
              viewBox="0 0 1024 1024"
              width="24px"
            >
              <title>
                arrowhead-down
              </title>
              <path
                d="M512 580.267l-226.133-226.133c-7.461-8.401-18.291-13.668-30.351-13.668-13.957 0-26.267 7.055-33.558 17.793l-0.091 0.142c-5.319 7.276-8.51 16.4-8.51 26.27 0 12.199 4.876 23.259 12.785 31.338l-0.008-0.008 256 253.867c7.434 8.083 18.061 13.13 29.867 13.13s22.432-5.047 29.84-13.101l0.026-0.029 256-253.867c7.901-8.071 12.777-19.131 12.777-31.33 0-9.87-3.192-18.994-8.599-26.397l0.088 0.127c-7.382-10.88-19.691-17.934-33.649-17.934-12.060 0-22.89 5.267-30.315 13.626l-0.036 0.042z"
              />
            </svg>
          </div>
          <div
            class="message"
            data-testid="message"
          />
        </div>
      </div>
      <div
        class="emotion-12 denali-input emotion-13 emotion-14"
        data-testid="input-wrapper"
      >
        <input
          autocomplete="off"
          data-testid="input-node"
          placeholder="Enter value"
          value=""
        />
        <div
          class="message"
          data-testid="message"
        />
      </div>
      <div
        class="emotion-15 emotion-16"
      >
        <svg
          class="emotion-17"
          data-testid="icon"
          height="1.75em"
          viewBox="0 0 1024 1024"
          width="1.75em"
        >
          <title>
            add-circle
          </title>
          <path
            d="M512 42.667c-259.206 0-469.333 210.128-469.333 469.333s210.128 469.333 469.333 469.333c259.206 0 469.333-210.128 469.333-469.333v0c0.003-0.635 0.005-1.387 0.005-2.138 0-258.027-209.173-467.2-467.2-467.2-0.752 0-1.504 0.002-2.255 0.005l0.117-0zM512 896c-212.077 0-384-171.923-384-384s171.923-384 384-384c212.077 0 384 171.923 384 384v0c-1.203 211.592-172.408 382.797-383.885 383.999l-0.115 0.001z"
          />
          <path
            d="M725.333 469.333h-170.667v-170.667c0-23.564-19.103-42.667-42.667-42.667s-42.667 19.103-42.667 42.667v0 170.667h-170.667c-23.564 0-42.667 19.103-42.667 42.667s19.103 42.667 42.667 42.667v0h170.667v170.667c0 23.564 19.103 42.667 42.667 42.667s42.667-19.103 42.667-42.667v0-170.667h170.667c23.564 0 42.667-19.103 42.667-42.667s-19.103-42.667-42.667-42.667v0z"
          />
        </svg>
      </div>
    </div>
    <div
      class="emotion-4 emotion-5"
    >
      <label
        class="emotion-6 denali-input-label emotion-7 emotion-8"
      />
      <div
        class="emotion-23 emotion-24"
      />
    </div>
  </div>
</div>
`;
